src/io/unix/sync.rs

```diff
@@ -156,7 +163,14 @@ impl File {
 
     pub(crate) async fn fsyncdata(&self) -> IOResult<()> {
         let file_inner = self.inner.clone();
-        Self::background_sync_call(move || file_inner.std_file.sync_all()).await
+        let size = self.size();
+        Self::background_sync_call(
+            move || {
+               file_inner.std_file.sync_all()?;
+               file_inner.synced_size.store(size, Ordering::SeqCst);
```

Comment:
I suggest to use `fetch_max` here for safety
```suggestion
               file_inner.synced_size.fetch_max(size, Ordering::SeqCst);
```